### ISSUE TITLE
2.1.19 Disallow beer change once samples registration is closed

### ISSUE THREAD

**Issue by grmelacz** [geoffhumphrey/brewcompetitiononlineentry#1180]
Version: 2.1.19

Installation URL: https://soutez.mhmeeting.cz

Is your installation hosted on brewcompetition.com or brewcomp.com? No

Description of Issue:
We had a sample in this year's competition that was previously registered in a different category. I've printed the beer list the evening before the competition, after the samples registration was closed (closed in the morning that day). The next day, the beer registered under that number was changed by the brewer to a different one (different style) which caused some confusion in the sorting. We were surprised to find out the user was able to change the beer after the samples registration was closed.

Should this be allowed? Is this a bug? I do not think it should be allowed, or at least there should be a switch to disable that behaviour because this may lead to an error when the beer previously registered as one style will be put to a wrong style batch to judge.

**Comment by geoffhumphrey** [comment]
For all participants without Admin rights, there's coding in place to prevent them to edit their entries after the prescribed entry window has closed. I tried several scenarios, but couldn't find any "back door" - perhaps the person in question was an Admin (Admins are able to edit any entry, including their own at any time) or one of your Admins changed the style?

**Comment by jimmyorkin** [comment]
I know I had this issue several years ago and my fix was to schedule a judging session at the end of the registration period.

**Comment by grmelacz** [comment]
We've finally found the beer list printed before the judging, after the samples registration was closed. It was, indeed the same as the current beer list. So it must have been a human error after all. There are some beers updated before the competition, after the registration has been closed, but those have to be my edits to reflect changes submitted by the contestants or based on the beer delivered.
